tests/types/test_schema.py

```diff
@@ -1636,6 +1634,59 @@ def test_schema_inference_on_lists(data, data_type):
     assert inferred_schema == expected_schema
 
 
+def test_schema_inference_with_empty_lists():
+    # If (non-nested) empty list is passed, should be inferred as string for backward compatibility
+    # with version 2.7.1: https://github.com/mlflow/mlflow/blob/v2.7.1/mlflow/types/utils.py#L150
+    data = []
+    assert _infer_schema(data) == Schema([ColSpec(DataType.string)])
+
+    # Nested list contains only an empty list is not allowed.
+    data = [[]]
```

Comment:
Is `[[None]]` not allowed either?

Reply:
No, similarly to empty list. Adding test case for it.